src/components/GoalsComponents/GoalsForm.tsx

```diff
@@ -78,15 +83,28 @@ export const GoalsForm = ({ selectedColorIndex }: { selectedColorIndex: number }
       setError("Enter a goal title! (P.S. the problem could also be with the sequence of goal-duration-frequency)");
       return;
     }
-    const newGoal = createGoal(goalTitle, goalFrequency === "daily", Number(goalDuration.split(" ")[0]), null, null, 0);
-    await addGoal(newGoal);
+    const newGoal = createGoal(
+      goalTitle,
+      goalFrequency === "daily",
+      Number(goalDuration.split(" ")[0]),
+      null,
+      null,
+      0,
+      parentGoalId!
+    );
+    const newGoalId = await addGoal(newGoal);
+    if (parentGoalId) {
+      const parentGoal = await getGoal(parentGoalId);
+      const newSublist = parentGoal?.sublist === undefined ? [newGoalId] : [...parentGoal?.sublist, newGoalId];
```

Comment:
you already checked whether the parentGoal.sublist is defined or not so while executing the block why use the '?' . I think that's why it is failing the tc

Reply:
not holding this PR. Quite amount of commits I c. I m Pushing the requested changes

Reply:
Right, that makes so much sense. I was using optional chaining earlier without the conditional statement so it was working, but now that isn't really needed, thanks for the commits